eclipse-scout-core/test/form/fields/smartfield/SmartFieldSpec.ts

```diff
@@ -604,6 +607,52 @@ describe('SmartField', () => {
       expect(displayText).toBe('Baz');
     });
 
+    it('lookupByKey should set a validation warning status if there was an error during processing', () => {
+      let field = _createFieldWithNoDataKeyLookupCall(true);
+      field._formatValue(4 /* non-existing key */); // triggers lookup by key
+      jasmine.clock().tick(500);
+      expect(field.errorStatus).toBeInstanceOf(ValidationFailedStatus);
+      expect(field.errorStatus.isWarning()).toBeTrue();
+    });
+
+    it('lookupByKey should set a validation warning status if the lookup call returns no lookup row', () => {
+      let field = _createFieldWithNoDataKeyLookupCall(false);
+
+      field._formatValue(0); // triggers lookup by key
+      jasmine.clock().tick(500);
+      expect(field.errorStatus).toBeInstanceOf(ValidationFailedStatus);
+      expect(field.errorStatus.isWarning()).toBeTrue();
+    });
+
+    function _createFieldWithNoDataKeyLookupCall(rejectPromise: boolean): SpecSmartField {
```

Comment:
There is no need to prefix the function with _ because it cannot be called from outside anyway. We use that pattern only for class methods because they can be accessed.
If you use setValue above, you can return SmartField<number> instead of SpecSmartField. In that case you need to change the scout.create slighly: return scout.create((SmartField<number>)
Please remove the 3 casts , they are not necessary.